flycheck.el

```diff
@@ -5813,16 +5814,30 @@ https://github.com/rust-lang/rust/blob/master/src/libsyntax/json.rs#L67-L139"
          errors)))
 
     ;; Then we turn children messages into flycheck errors pointing to the
-    ;; location of the primary span.  According to the format, children
-    ;; may contain spans, but they do not seem to use them in practice.
+    ;; location of the primary span.
     (dolist (child children)
       (let-alist child
         (push
          (flycheck-error-new-at
-          primary-line
-          primary-column
+          ;; Use the line/column from the first span if there is one, or
+          ;; fallback to the line/column information from the primary span of
+          ;; the diagnostic.
+          (or (and (car .spans)
+                   (cdr (assq 'line_start (car .spans))))
+              primary-line)
+          (or (and (car .spans)
+                   (cdr (assq 'column_start (car .spans))))
+              primary-column)
           'info
-          .message
+          ;; Messages from `cargo clippy` may suggest replacement code.  In
+          ;; these cases, the `message` field itself is an unhelpful `try` or
+          ;; `change this to`.  We add the `suggested_replacement` field in
+          ;; these cases.
+          (-if-let* ((first-span (car .spans))
+                     (replacement
+                      (cdr (assq 'suggested_replacement first-span))))
+              (format "%s: `%s`" .message replacement)
```

Comment:
It might be better to use `` `…' `` instead of `` `…` `` here?

Reply:
I believe `rustc` uses the Markdown convention, so it might be more coherent to follow that in the displayed message.  It's just text, so it doesn't really matter.  Unless someone tries to prettify that down the road.

Reply:
That's what I'm afraid about (prettification).  I think popups do get prettified, though arguably it's a bug on the Emacs side.